### PR TITLE
Update Clippy

### PR DESCRIPTION
**Pull request by Xanewok**
Should fix the fallout caused by https://github.com/rust-lang/rust/pull/58321 (still testing it locally).

r? @oli-obk

### CODE REVIEW

**Comment by Xanewok** [comment]
cc #58765

**Comment by Xanewok** [comment]
https://github.com/rust-lang/rust/pull/58754 also contains the relevant clippy update when/if that lands

**Comment by ljedrz** [comment]
@Xanewok #58754 has a slightly bumped priority, so it should land soon (it's currently 5th in the queue).

**Comment by kennytm** [comment]
#58754 has been merged.

**Comment by ljedrz** [comment]
@kennytm unfortunately without the clippy update; now https://github.com/rust-lang/rust/pull/58836 has one, but it might need another branch due to tidy not allowing the custom compiletest-rs branch.

TL;DR: I think this PR should be merged

**Comment by Xanewok** [comment]
If the API is meant to be 1-1 compatible, maybe it'd make sense to patch the `compiletest-rs` to point to https://github.com/rust-lang/rust/tree/master/src/tools/compiletest (just like we do with Rustfmt and clippy in-tree)? I think this should unblock the releases until https://github.com/laumann/compiletest-rs/pull/160 is merged and a new version is released.

**Comment by mati865** [comment]
@Xanewok I think that's good idea, tools upgrades would no longer be blocked when some changes break `compiletest-rs`.

**Comment by Xanewok** [comment]
`compiletest` is [fixed](https://github.com/laumann/compiletest-rs/pull/160) now, bumping Clippy to include the fix updating this dependency: https://github.com/rust-lang/rust-clippy/pull/3838

**Comment by Xanewok** [comment]
Tested locally and both Clippy and RLS successfully compile now.
Bumping prio in hope of it making to the next nightly.

r? @Xanewok 
@bors r+ p=1
